workflow-manager/batchpath/batchpath.go

```diff
@@ -0,0 +1,133 @@
+package batchpath
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/prio-server/workflow-manager/utils"
+)
+
+type BatchPath struct {
+	AggregationID  string
+	dateComponents []string
+	ID             string
+	Time           time.Time
+	metadata       bool
+	avro           bool
+	sig            bool
+}
+
+type List []*BatchPath
+
+func (bpl List) Len() int {
+	return len(bpl)
+}
+
+func (bpl List) Less(i, j int) bool {
+	return bpl[i].Time.Before(bpl[j].Time)
+}
+
+func (bpl List) Swap(i, j int) {
+	bpl[i], bpl[j] = bpl[j], bpl[i]
+}
+
+func New(batchName string) (*BatchPath, error) {
```

Comment:
nit: exported methods should have doc strings. I know the original code lacked substantial comments or docs, but since you're going to the trouble of cleaning this up, now might be the time.

Reply:
Yep! I should add those comments. Thanks for reminding me!